src/main/java/com/google/devtools/build/lib/runtime/TestSummaryPrinter.java

```diff
@@ -184,12 +189,42 @@ public static void print(
 
           if (summary.getFailedTestCasesStatus() != FailedTestCasesStatus.FULL) {
             terminalPrinter.print(
+                    Mode.WARNING
+                            + "    (some shards did not report details, list of failed test"
+                            + " cases incomplete)\n"
+                            + Mode.DEFAULT);
+          }
+        }
+      }
+    } else if (showAllTestCasesDetailed) {
```

Comment:
this method became pretty big, maybe split each if case to separate private method